client/state/notices/account-recovery/index.js

```diff
@@ -6,9 +6,43 @@ import { translate } from 'i18n-calypso';
 /**
  * Internal dependencies
  */
-import { errorNotice } from 'state/notices/actions';
+import {
+	successNotice,
+	errorNotice,
+} from 'state/notices/actions';
+
+import {
+	getAccountRecoveryPhone,
+	getAccountRecoveryEmail,
+} from 'state/account-recovery/settings/selectors';
+
 import { dispatchError } from '../utils';
 
+const getUpdateSuccessMessage = ( target, getState ) => {
+	switch ( target ) {
+		case 'phone':
+			const oldPhone = getAccountRecoveryPhone( getState() );
+
+			if ( null == oldPhone ) {
+				return translate( 'Successfully added new recovery SMS number.' );
+			}
+
+			return translate( 'Successfully updated recovery SMS number.' );
+
+		case 'email':
+			const oldEmail = getAccountRecoveryEmail( getState() );
+
+			if ( '' === oldEmail ) {
```

Comment:
Since it looks like we are agreeing on taking a null as the absence of value, there seems also no need for the reducer to convert it to an empty string any more. I've updated both the check and the reducer in 5582998c4a396804fec98c245a902d1de8fc0d4e.